### PR TITLE
Allow passing backend classes from CLI and other APIs

### PR DESCRIPTION
**Pull request by jezdez**
This is related to https://github.com/jezdez/Flask-RQ2/issues/2 and https://github.com/jezdez/Flask-RQ2/issues/12. While debugging those issues I've found many inconsistencies how the backend classes are used throughout the RQ code base, basically making it currently impossible to fully override job, queue, worker or connection classes.

This should be backward-compatible, but any testing in real-world projects would be appreciated.

### CODE REVIEW

File: tests/test_worker.py [hunk 1]
```diff
@@ -626,7 +626,7 @@ class TimeoutTestCase:
     def setUp(self):
         # we want tests to fail if signal are ignored and the work remain
         # running, so set a signal to kill them after X seconds
-        self.killtimeout = 10
+        self.killtimeout = 15
```
Comment (jezdez) [inline tests/test_worker.py]: FYI this was the only way I was able to fix a slow subprocess call in the pypy env on Travis CI.

File: rq/job.py [hunk 2]
```diff
@@ -123,7 +124,7 @@ def create(cls, func, args=None, kwargs=None, connection=None,
             job._instance = func
             job._func_name = '__call__'
         else:
-            raise TypeError('Expected a callable or a string, but got: {}'.format(func))
+            raise TypeError('Expected a callable or a string, but got: {0}'.format(func))
```
Comment (jezdez) [inline rq/job.py]: Change needed to work on Python 2.6.

**Comment by selwin** [comment]
Wow, this is a big PR. I'll need some time to review this. If possible, next time please split them into smaller, logically grouped PRs :)

**Comment by selwin** [comment]
Thanks for the PR @jezdez !

**Comment by jezdez** [comment]
@selwin Happy to :)

**Pull request merged**